docs/src/tutorials/nonlinear/classifiers.jl

```diff
@@ -0,0 +1,274 @@
+println("Minimum slack: ", minimum(y_sol), "\nMaximum slack: ", maximum(y_sol))
+
+# With the solution, we can ask: was the value of the penalty constant
+# "sufficiently large" for this data set? This can be judged in part by the
+# range of the slack variables. If the slack is too large, then we need to
+# increase the penalty constant.
+
+# Let's plot the solution and check how we did:
+
+Plots.plot!(
+    plot,
+    x -> line(x; w = w_sol, g = g_sol),
+    0.0:0.01:2.0;
+    seriestype = :line,
+    linewidth = 5,
+    linestyle = :dashdotdot,
+    lineopacity = 0.9,
+    c = :darkblue,
+)
+
+# We find that we have recovered the dividing line from just the information of
+# the points and their labels.
+
+# ## Nonseparable classes of points
+
+# Now, what if the point sets are not cleanly separable by a line
+# (or a hyperplane in higher dimensions)? Does this still work?
+# Let's repeat the above process but this time we will simulate nonseparable
+# classes of points by intermingling a few nearby points across the previously used line.
+
+nearby_indices = (abs.(line.(eachrow(P))) .< 1.1)
+labels_new = ifelse.(nearby_indices, -labels, labels)
+
+P_pos = P[labels_new.==1, :]
+P_neg = P[labels_new.==-1, :]
+@assert size(P_pos, 1) + size(P_neg, 1) == size(P, 1) #src
+Plots.scatter!(
+    plot_copy,
+    P_pos[:, 1],
+    P_pos[:, 2];
+    shape = :cross,
+    markercolor = :blue,
+    markersize = 8,
+)
+Plots.scatter!(
+    plot_copy,
+    P_neg[:, 1],
+    P_neg[:, 2];
+    shape = :xcross,
+    markercolor = :crimson,
+    markersize = 8,
+)
+
+model, w_sol, g_sol, y_sol = solve_SVM_classifier(P, labels_new)
+Test.@test termination_status(model) == LOCALLY_SOLVED    #src
+Test.@test primal_status(model) == FEASIBLE_POINT  #src
+println("Minimum slack: ", minimum(y_sol), "\nMaximum slack: ", maximum(y_sol))
+
+Plots.plot!(
+    plot_copy,
+    x -> line(x; w = w_sol, g = g_sol),
+    0.0:0.01:2.0;
+    seriestype = :line,
+    linewidth = 5,
+    linestyle = :dashdotdot,
+    lineopacity = 0.9,
+    c = :darkblue,
+)
+
+# So it still works as a classifier, with allowance for the nonseparable points.
+#
+# We can find out which points are contributing to the shape of the line by
+# looking at the dual values of the affine constraints and comparing them
+# to the penalty constant ``C``:
+
+affine_cons = all_constraints(model, AffExpr, MOI.GreaterThan{Float64})
+active_cons = findall(isapprox.(dual.(affine_cons), C_0; atol = 0.001))
+
+findall(nearby_indices) ⊆ active_cons
+
+# The last statement tells us that our nonseparable points are actively contributing
+# to how the classifier is defined. The remaining points are of interest and
+# are highlighted:
+
+P_active = P[setdiff(active_cons, findall(nearby_indices)), :]
+
+Plots.scatter!(
+    plot_copy,
+    P_active[:, 1],
+    P_active[:, 2];
+    shape = :hexagon,
+    markercolor = :green,
+    markersize = 8,
+    markeropacity = 0.5,
+)
+
+# ## Advanced: Duality and the kernel method
```

Comment:
Are you planning to add more here? It could probably end above here and be a good tutorial. I don't know if we need more?

Reply:
Yes, but I might have gone overboard a bit. Could possibly be split into two.